configs/module_test.go

```diff
@@ -99,6 +100,62 @@ func TestNewModule_resource_providers(t *testing.T) {
 	}
 }
 
+func TestModule_required_providers_multiple(t *testing.T) {
```

Comment:
I think it could be helpful to have a comment here that describes the original issue, where the order of the files being appended matter.

Reply:
Good point, will do!